### ISSUE TITLE
FullScalaVersion chokes on Scala nightlies

### ISSUE THREAD

**Issue by SethTisue** [djspiewak/sbt-spiewak#5]
nightlies have version numbers of the form `2.13.2-bin-7bfe4c0 ` (the `-bin-` might instead be `-pre-`), but that isn't accepted

this came up as I tried to add cats-effect-testing to the Scala community build

```
sbt:cats-effect-testing> ++2.13.2-bin-7bfe4c0!
[info] Forcing Scala version to 2.13.2-bin-7bfe4c0 on all projects.
[info] Reapplying settings...
[error] scala.MatchError: 2.13.2-bin-7bfe4c0 (of class java.lang.String)
[error] Use 'last' for the full log.
sbt:cats-effect-testing> last
[debug] > Exec(++2.13.2-bin-7bfe4c0!, Some(28205be9-3231-4de5-a0ae-8f10cd463119), Some(CommandSource(console0)))
[debug] Evaluating tasks: checkBuildSources
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
[info] Forcing Scala version to 2.13.2-bin-7bfe4c0 on all projects.
[debug] Switching Scala version on:
[debug]     scalatest (2.12.10, 2.13.1)
[debug]     scalatest-scalacheck (2.12.10, 2.13.1)
[debug]   * root (2.12.10, 2.13.1)
[debug]     minitest (2.12.10, 2.13.1)
[debug]     utest (2.12.10, 2.13.1)
[debug]     specs2 (2.12.10, 2.13.1)
[debug] Excluding projects:
[info] Reapplying settings...
[error] scala.MatchError: 2.13.2-bin-7bfe4c0 (of class java.lang.String)
[error] 	at sbtspiewak.FullScalaVersion$.unapply(FullScalaVersion.scala:28)
[error] 	at sbtspiewak.SpiewakPlugin$.$anonfun$projectSettings$11(SpiewakPlugin.scala:213)
[error] 	at sbt.internal.util.EvaluateSettings$MixedNode.evaluate0(INode.scala:223)
[error] 	at sbt.internal.util.EvaluateSettings$INode.evaluate(INode.scala:166)
[error] 	at sbt.internal.util.EvaluateSettings.$anonfun$submitEvaluate$1(INode.scala:87)
[error] 	at sbt.internal.util.EvaluateSettings.sbt$internal$util$EvaluateSettings$$run0(INode.scala:99)
[error] 	at sbt.internal.util.EvaluateSettings$$anon$3.run(INode.scala:94)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] scala.MatchError: 2.13.2-bin-7bfe4c0 (of class java.lang.String)
[error] Use 'last' for the full log.
[debug] > Exec(shell, None, None)
[debug] Evaluating tasks: checkBuildSources
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
sbt:cats-effect-testing> 
```

**Comment by djspiewak** [comment]
Rip regex fail. Is that always the exact pattern? Should be easy enough for me to fix.

**Comment by djspiewak** [comment]
Fixed in master, I think. I've updated cats-effect-testing. Lmk.

**Comment by SethTisue** [comment]
thanks Daniel! giving it a try over at https://github.com/scala/community-build/pull/1052

**Comment by SethTisue** [comment]
well that worked, but I found another problem I should have seen coming but didn't

another version number format that it still chokes on is e.g. `2.13.2-bin-b7d9d19-SNAPSHOT` — this is the format of a Scala version built for an as-yet-unmerged PR

here's the log: https://scala-ci.typesafe.com/view/scala-2.13.x/job/scala-2.13.x-integrate-community-build/3080/artifact/logs/cats-effect-testing-extract.log

**Comment by dwijnand** [comment]
> Is that always the exact pattern?

https://github.com/sbt/librarymanagement/blob/3132a3eb59fcfbabf33683388377b39a5458457d/core/src/main/scala/sbt/librarymanagement/CrossVersionExtra.scala#L61

**Comment by SethTisue** [comment]
ping @djspiewak, might you have time to tweak this one more time?

**Comment by djspiewak** [comment]
@SethTisue Thanks for the re-ping! I'll try to get to it tonight (us time).

**Comment by djspiewak** [comment]
Fixed in 0.13.0. @SethTisue @dwijnand Sorry for the delay!
